pyvista/themes.py

```diff
@@ -0,0 +1,2071 @@
+"""API description for managing plotting theme parameters in pyvista.
+
+Examples
+--------
+Apply a built-in theme
+
+>>> import pyvista
+>>> pyvista.set_plot_theme('default')
+>>> pyvista.set_plot_theme('document')
+>>> pyvista.set_plot_theme('dark')
+>>> pyvista.set_plot_theme('paraview')
+
+Load a theme into pyvista
+
+>>> theme = pyvista.themes.DefaultTheme()
+>>> theme.save('my_theme.json')  # doctest:+SKIP
+>>> loaded_theme = pyvista.load_theme('my_theme.json')  # doctest:+SKIP
+
+Create a custom theme from the default theme and load it into
+pyvista.
+
+>>> my_theme = pyvista.themes.DefaultTheme()
+>>> my_theme.font.size = 20
+>>> my_theme.font.title_size = 40
+>>> my_theme.cmap = 'jet'
+...
+>>> pyvista.global_theme.load_theme(my_theme)
+>>> pyvista.global_theme.font.size
+20
+
+"""
+
+import json
+from typing import Union, List
+import warnings
+from enum import Enum
+import os
+
+from .plotting.colors import PARAVIEW_BACKGROUND, get_cmap_safe
+from .plotting.tools import parse_color, parse_font_family
+from .utilities.misc import PyvistaDeprecationWarning
+from .core.errors import DeprecationError
+
+
+class _rcParams(dict):  # pragma: no cover
+    """Reference to the deprecated rcParams dictionary."""
+
+    def __getitem__(self, key):
+        raise DeprecationError('rcParams is deprecated.  Please use ``pyvista.global_theme``')
+
+    def __setitem__(self, key, value):
+        raise DeprecationError('rcParams is deprecated.  Please use ``pyvista.global_theme``')
+
+
+def load_theme(filename):
+    """Load a theme from a file.
+
+    Examples
+    --------
+    >>> import pyvista
+    >>> theme = pyvista.themes.DefaultTheme()
+    >>> theme.save('my_theme.json')  # doctest:+SKIP
+    >>> loaded_theme = pyvista.load_theme('my_theme.json')  # doctest:+SKIP
+
+    """
+    return DefaultTheme.from_dict(json.load(open(filename)))
+
+
+def set_plot_theme(theme):
+    """Set the plotting parameters to a predefined theme using a string.
+
+    Parameters
+    ----------
+    theme : str
+        Theme name.  Either ``'default'``, ``'document'``, ``'dark'``,
+        or ``'paraview'``.
+
+    Examples
+    --------
+    Set to the default theme.
+
+    >>> import pyvista
+    >>> pyvista.set_plot_theme('default')
+
+    Set to the document theme.
+
+    >>> pyvista.set_plot_theme('document')
+
+    Set to the dark theme.
+
+    >>> pyvista.set_plot_theme('dark')
+
+    Set to the ParaView theme.
+
+    >>> pyvista.set_plot_theme('paraview')
+
+    """
+    import pyvista
+    if isinstance(theme, str):
+        theme = theme.lower()
+        if theme == 'night':  # pragma: no cover
+            warnings.warn('use "dark" instead of "night" theme', PyvistaDeprecationWarning)
+        new_theme = _ALLOWED_THEMES[theme].value()
```

Comment:
Tricky. I first thought this was missing a function call, but the Enum has a `.value` property. Might not be misleading enough to merit binding the class to a separate name first.

Reply:
> Might not be misleading enough to merit binding the class to a separate name first.

How would you implement this?

Reply:
I just meant doing
```py
new_theme_type = _ALLOWED_THEMES[theme].value
pyvista.global_theme.load_theme(new_theme_type())
```
or something to that effect. The implementation is fine as-is, it just challenged my mental parser :)

Reply:
I thought about this a little bit too (how it's confusing that we're calling `value`).  It makes more sense if we return the type and instantiate it as you suggested.